### PR TITLE
Remove Accept header from thecatapi request

### PR DESCRIPTION
**Pull request by fejta**
/assign @ixdy @michelle192837 @BenTheElder 

/priority critical-urgent
/bark

### CODE REVIEW

**Comment by fejta** [comment]
>  fejta requested review from matthyx and removed request for cjwagner and stevekuznetsov just now

Removal wasn't intentional, just clicked the request button

File: prow/plugins/cat/cat.go [hunk 1]
```diff
@@ -143,26 +141,27 @@ func (r *realClowder) Url(category string) string {
...
+	if a.Image == "" {
+		return "", fmt.Errorf("no image url in response from %s", uri)
+	}
 	// checking size, GitHub doesn't support big images
 	toobig, err := github.ImageTooBig(a.Image)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("could not validate image size %s: %v", a.Image, err)
 	} else if toobig {
-		return "", errors.New("unsupported cat :( size too big: " + a.Image)
```
Comment (stevekuznetsov) [inline prow/plugins/cat/cat.go]: :(

Comment (ixdy) [reply prow/plugins/cat/cat.go]: please update this error message to "long cat is too long"

**Comment by fejta** [comment]
Given that `http.Get()` is WAY easier than the client, accept header shenanigans. I am pretty sure the api previously required it.

File: prow/plugins/cat/cat.go (hunk 1 above)
Comment (BenTheElder) [reply prow/plugins/cat/cat.go]: /hold
this is an important change

File: prow/plugins/cat/cat.go [hunk 2]
```diff
@@ -143,26 +141,27 @@ func (r *realClowder) Url(category string) string {
...
-	req.Header.Add("Accept", "application/json")
-	resp, err := client.Do(req)
+	resp, err := http.Get(uri)
 	if err != nil {
 		return "", fmt.Errorf("could not read cat from %s: %v", uri, err)
 	}
+	if sc := resp.StatusCode; sc > 299 || sc < 200 {
+		return "", fmt.Errorf("failing %d response from %s", sc, uri)
+	}
 	defer resp.Body.Close()
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/plugins/cat/cat.go]: Done

File: prow/plugins/cat/cat.go (hunk 1 above)
Comment (fejta) [reply prow/plugins/cat/cat.go]: Done

**Comment by matthyx** [comment]
@fejta I obviously don't know what was wrong with the `http.Client{}` in that plugin but could you check also `prow/plugins/dog/dog.go` as there is exactly the same code there?

**Pull request merged**